src/main/java/server/Server.java

```diff
@@ -139,150 +136,142 @@ private JSONObject parseDebugOutput(String line) {
       obj.put("submodule", matcher.group(3));
       obj.put("log", matcher.group(4));
     }
-    
+
     obj.put("line", line);
     return obj;
   }
 
-  //Compiles the pod code
+  // Compiles the pod code
   public void debugCompile() {
     initDebugData();
     debugStatus = COMPILING;
-    isCompiling = true; // The compiling process start
-
-    String DIR_PATH = FileSystems.getDefault().getPath("./").toAbsolutePath().toString();
-    String HYPED_PATH = DIR_PATH.substring(0, DIR_PATH.length() - 1) + "hyped-pod_code";
-
-    ArrayList<String> command = new ArrayList<String>();
-    // Suppose to be make -j, but it's so buggy so leave it as make for now
-    command.add("make");
-
+    isCompiling = true;
+    final Path buildDirectoryPath = Paths.get(BUILD_DIRECTORY);
+    final String cmakeFlags = "-DPEDANTIC=ON -DRELEASE=ON -DCOVERAGE=ON -DCROSS=OFF -DCMAKE_CXX_COMPILER=/usr/bin/clang++";
+    if (!Files.isDirectory(buildDirectoryPath)) {
+      System.out.printf("Did not find build directory at %s\n", BUILD_DIRECTORY);
+      try {
+        System.out.printf("Creating build directory at %s\n", BUILD_DIRECTORY);
+        Files.createDirectories(buildDirectoryPath);
+      } catch (final IOException e) {
+        System.out.printf("Failed to create build directory at %s\n", BUILD_DIRECTORY);
+        e.printStackTrace();
+        isCompiling = false;
+        debugData.put(IS_COMPILED, false);
+        debugData.put(IS_SUCCESS, false);
+        convertDebugOutput();
+        return;
+      }
+    }
+    Runtime runtime = Runtime.getRuntime();
     try {
-      System.out.println("Compiling from: " + HYPED_PATH);
-      debugOutput = new JSONArray();
-      compileProcess = new ProcessBuilder(command).directory(new File(HYPED_PATH)).start();
-
-      StreamGobblerCompile errorGobbler = new StreamGobblerCompile(compileProcess.getErrorStream());
-      StreamGobblerCompile outputGobbler = new StreamGobblerCompile(compileProcess.getInputStream());
-          
-      // start gobblers
-      outputGobbler.start();
+      String cmakeCommand = String.format("cmake -S . -B %s %s", BUILD_DIRECTORY, cmakeFlags);
+      System.out.printf("Running '%s'\n", cmakeCommand);
+      Process cmakeProcess = runtime.exec(cmakeCommand);
+      StreamGobblerCompile errorGobbler = new StreamGobblerCompile(cmakeProcess.getErrorStream());
+      StreamGobblerCompile outputGobbler = new StreamGobblerCompile(cmakeProcess.getInputStream());
       errorGobbler.start();
-
-      compileProcess.waitFor();
-
-      boolean isSuccess = compileProcess.exitValue() == 0;
-
-      //Procedure to check wheter hyped code is successfully compiled
-      boolean isFileExisted = isHypedExist(); // For first time compilation
-
+      outputGobbler.start();
+      cmakeProcess.waitFor();
       debugData.put(IS_COMPILED, true);
```

Comment:
Was compiling but not showing run button, so I looked through the changes for any missing lines and found this one. Added it and now when compiling is done it shows run/recompile buttons. @MiltFra